bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -279,6 +262,59 @@ func (r *KubeadmConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	return r.joinWorker(ctx, scope)
 }
 
+func (r *KubeadmConfigReconciler) refreshBootstrapToken(ctx context.Context, config *bootstrapv1.KubeadmConfig, cluster *clusterv1.Cluster) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+	token := config.Spec.JoinConfiguration.Discovery.BootstrapToken.Token
+
+	remoteClient, err := r.remoteClientGetter(ctx, r.Client, util.ObjectKey(cluster))
+	if err != nil {
+		log.Error(err, "Error creating remote cluster client")
+		return ctrl.Result{}, err
+	}
+
+	log.Info("Refreshing token until the infrastructure has a chance to consume it")
+	err = refreshToken(ctx, remoteClient, token)
+	if err != nil {
+		// It would be nice to re-create the bootstrap token if the error was "not found", but we have no way to update the Machine's bootstrap data
```

Comment:
🤔 I'd probably remove this comment, if the token is not found we should definitely not create it again, but wait for intervention from a health check or operator

Reply:
This whole function was extracted exactly as-is from https://github.com/kubernetes-sigs/cluster-api/pull/3940/files#diff-f754cabdd737c0fac3039e7eb563b50ea52fc4ff88a9d517f4067e25977acdadL233-L251. I didn't touch the pre-existing comments since I wasn't making any changes to the refresh behavior.

Reply:
Got it. These are a bit misleading and I'd remove them if you're ok with it

Reply:
will do